1단계 미션 제출합니다.

chanhalee opened the pull request:
안녕하세요 chanhalee 입니다.
1단계 미션 제출합니다.
잘 부탁드립니다!

findstar commented on src/main/java/db/DataBase.java:
```diff
@@ -6,9 +6,11 @@
 
 public class DataBase {
     private static Map<String, User> users = Maps.newHashMap();
+    private static Map<String, HttpSession> sessions = Maps.newHashMap();
```
Database에 HttpSession 이 존재하는 구조를 의도하신거라면 별도로 분리하는게 어떨까합니다. 
실제 웹서비스에서도 DBSession 을 사용하기도 하지만, 
사용자의 정보를 저장하는 용도의 Database 와 
Session 스토리지를 분리하는게 책임을 나눌 수 있는 방법 같아보입니다.

findstar commented on src/main/java/webserver/RequestHandler.java:
```diff
@@ -38,6 +39,7 @@ public void run() {
         try (InputStream in = connection.getInputStream(); OutputStream out = connection.getOutputStream()) {
             HttpRequest request = HttpRequest.from(in);
             HttpResponse response = new HttpResponse(out);
+            response.setCookie(HttpSession.SESSION_COOKIE_NAME, request.getSession().getId());
```
적절한 상수처리 👍

findstar commented on src/main/java/webserver/model/HttpRequest.java:
```diff
@@ -36,13 +36,14 @@ public static HttpRequest from(InputStream in) throws IOException {
 
         if (method == HttpMethod.POST || method == HttpMethod.PUT) {
-            query = IOUtils.readData(reader, Integer.parseInt(headers.getOrDefault("content-length", "0")));
+            String contentQuery = IOUtils.readData(reader, Integer.parseInt(headers.getOrDefault("content-length", "0")));
```
이 코드는 메서드 호출의 깊이가 너무 깊어지는데요 적절하게 extract 하면 어떨까요?

findstar commented on src/test/java/webserver/model/HttpRequestTest.java:
```diff
@@ -29,15 +32,45 @@ void parameters(int expected, String requestLine) throws IOException {
+    }
+
+    @Test
+    void queryParamsWithContent() throws IOException {
```
테스트 보강해주셨네요 💯

findstar reviewed (changes requested):
리뷰가 늦어 죄송합니다 ㅠ_ㅠ 🙏
마지막 단계인 세션을 잘 구현해주셨고 테스트 코드도 보강해주셨네요.
세션관련 코멘트 남겨놓았습니다. 요거 하나만 다시 확인 부탁드립니다  😁

chanhalee committed:
docs(README) Todo List 작성

Todo List 작성

chanhalee committed:
feat(HttpRequest) HttpRequest 작성 및 테스트 작성

HttpRequest 작성 및 테스트 작성

chanhalee committed:
feat(RequestHandler) HttpRequest 생성에 따른 RequestHandler 역할 축소

HttpRequest 생성에 따른 RequestHandler 역할 축소

chanhalee committed:
feat(HttpResponse) HttpResponse 작성

HttpResponse 작성

chanhalee committed:
feat(Controller) Controller 작성

Controller 작성
- Request 성격에 따라 적절한 Controller를 생성해 반환하는 RequestMapping 작성
- 홈 경로에 보낸 Get 요청에 대해 200 OK를 보내는 DefaultController 작성
- index.html, style.css 등 리소스 청구요청을 처리하는 컨트롤러 TemplateController 작성
- User 생성 요청을 처리하는 UserController 작성
